extensions/native/compiler/src/conversion/mod.rs

```diff
@@ -558,45 +349,159 @@ fn convert_instruction<F: PrimeField32, EF: ExtensionField<F>>(
                 AS::Native,
                 AS::Native,
                 AS::Immediate
-            )]
-        } else {
-            panic!(
-                "Unsupported instruction {:?}, field arithmetic is disabled",
-                instruction
-            )
-        },
-        AsmInstruction::AddF(..)
-        | AsmInstruction::SubF(..)
-        | AsmInstruction::MulF(..)
-        | AsmInstruction::DivF(..)
-        | AsmInstruction::AddFI(..)
-        | AsmInstruction::SubFI(..)
-        | AsmInstruction::MulFI(..)
-        | AsmInstruction::DivFI(..)
-        | AsmInstruction::SubFIN(..)
-        | AsmInstruction::DivFIN(..) => {
-            if options.field_arithmetic_enabled {
-                convert_base_arithmetic_instruction(instruction, options)
-            } else {
-                panic!(
-                    "Unsupported instruction {:?}, field arithmetic is disabled",
-                    instruction
-                )
-            }
-        }
-        AsmInstruction::AddE(..)
-        | AsmInstruction::SubE(..)
-        | AsmInstruction::MulE(..)
-        | AsmInstruction::DivE(..) => {
-            if options.field_extension_enabled {
-                convert_field_extension(instruction, options)
-            } else {
-                panic!(
-                    "Unsupported instruction {:?}, field extension arithmetic is disabled",
-                    instruction
-                )
-            }
-        }
+            )],
+            AsmInstruction::AddF(dst, lhs, rhs) => vec![
+                // mem[dst] <- mem[lhs] + mem[rhs]
+                inst_med(
+                    options.opcode_with_offset(FieldArithmeticOpcode::ADD),
+                    i32_f(dst),
+                    i32_f(lhs),
+                    i32_f(rhs),
+                    AS::Native,
+                    AS::Native,
+                    AS::Native,
+                ),
+            ],
+            AsmInstruction::AddFI(dst, lhs, rhs) => vec![
+                // mem[dst] <- mem[lhs] + rhs
+                inst_med(
+                    options.opcode_with_offset(FieldArithmeticOpcode::ADD),
+                    i32_f(dst),
+                    i32_f(lhs),
+                    rhs,
+                    AS::Native,
+                    AS::Native,
+                    AS::Immediate,
+                ),
+            ],
+            AsmInstruction::SubF(dst, lhs, rhs) => vec![
```

Comment:
it would save a bit of code if you grouped AddF, SubF, MulF together? similarly for AddFI etc